pypeit/sciimgstack.py

```diff
@@ -0,0 +1,385 @@
+        else:
+            sciImg = self.proc_list('sci', reject_cr=True, sigma_clip=False, sigrej=sigrej, maxiters=maxiters)
+
+        # Show the science image if an interactive run, only show the crmask
+        if show:
+            # Only mask the CRs in this image
+            self.show(sciImg.image * (sciImg.crmask == 0), chname='sciimg')
+
+        #return self.sciimg, self.sciivar, self.rn2img, self.mask, self.crmask
+        return sciImg
+
+    def proc_list(self, ltype, reject_cr=True, sigma_clip=False, sigrej=None, maxiters=5):
+        """
+        Process a list of science images
+
+        This includes stacking the images if there is more than 1
+
+        Args:
+            ltype (str): Type of images to process ('sci', 'bkg')
+            reject_cr (bool, optional):
+            sigrej (int or float, optional): Rejection threshold for sigma clipping.
+                 Code defaults to determining this automatically based on the numberr of images provided.
+            maxiters (int, optional):
+            show (bool, optional):
+
+        Returns:
+            ndarray, ndarray, ndarray, ndarray, ndarray:
+              sciimg
+              sciivar
+              rn2img
+              mask
+              crmask
+
+        """
+        # Init
+        if ltype == 'sci':
+            files = self.file_list
+        elif ltype == 'bkg':
+            files = self.bg_file_list
+        else:
+            msgs.error("Bad ltype for proc_list")
+        nimg = len(files)
+        weights = np.ones(nimg)/float(nimg)
+
+        # Load
+        img_stack, ivar_stack, rn2img_stack, crmask_stack, mask_stack = self.build_stack(
+            files, self.sci_bpm, reject_cr=reject_cr)
+
+        # ToDO The bitmask is not being properly propagated here!
+        if nimg > 1:
+            img_list = [img_stack]
+            var_stack = utils.calc_ivar(ivar_stack)
+            var_list = [var_stack, rn2img_stack]
+            img_list_out, var_list_out, outmask, nused = coadd2d.weighted_combine(
+                weights, img_list, var_list, (mask_stack == 0),
+                sigma_clip=sigma_clip, sigma_clip_stack = img_stack, sigrej=sigrej, maxiters=maxiters)
+            '''
+            img = img_list_out[0]
+            ivar = utils.calc_ivar(var_list_out[0])
+            rn2img = var_list_out[1]
+            '''
+            sciImage = scienceimage.ScienceImage.from_images(self.spectrograph, self.det,
+                                                             self.par['process'], self.sci_bpm,
+                                                             img_list_out[0],
+                                                             utils.calc_ivar(var_list_out[0]),
+                                                             var_list_out[1], np.invert(outmask),
+                                                             files=files)
```

Comment:
As far as I can tell, ScienceImage is the only class that uses your ImageMask class. I don't see why you are using inheritance here (ScienceImage child of ImageMask) and creating the extra class (ImageMask) at all. Why are those methods not just part of ScienceImage. Using inheritance in this way, where only one class inherits the methods, just creates extra lines of code for no reason, and a never-ending hierarchy of wrappers that is extremely hard to follow.

Reply:
I only stripped it out because you asked for it.
You are welcome to put it back the way I had it.